errors.go

```diff
@@ -333,8 +295,111 @@ func buildError(desc string, fullText string, wrapped error, cause Error) *struc
 	} else {
 		e.data["error_text"] = cleanedDesc
 	}
-	e.data["error_type"] = errorType
+	e.data["error_type"] = "errors.Error"
+
+	return e
+}
+
+type topLevelPrinter interface {
+	// Returns a printer which prints only the top-level error and any associated stack trace. The
+	// output of this printer will be a prefix of the output from MultiLinePrinter().
+	topLevelPrinter() func(*bytes.Buffer) bool
+}
+
+type unwrapper interface {
+	Unwrap() error
+}
+
+type wrappingError struct {
+	*baseError
+	wrapped error
+}
+
+// Implements error unwrapping as described in the standard library's errors package:
+// https://golang.org/pkg/errors/#pkg-overview
+func (e *wrappingError) Unwrap() error {
+	return e.wrapped
+}
+
+func (e *wrappingError) Fill(m context.Map) {
+	type filler interface{ Fill(context.Map) }
+
+	if f, ok := e.wrapped.(filler); ok {
```

Comment:
What if you have one of our errors wrapped in a standard Go error? I think maybe we need to unwrap as far as we can and look for all `filler`s along the way.

Reply:
That's a good point.  Fixed here: https://github.com/getlantern/errors/pull/6/commits/ee0d19731faa77ba4bda428fdb87d722524cd52e

Reply:
This is cleaner I think: https://github.com/getlantern/errors/pull/6/commits/8fa0b61f584e2d32c84c076ff76da071430aac68